ports/mynewt/src/memfault_platform_port.c

```diff
@@ -0,0 +1,142 @@
+  // !FIXME: Populate with platform device information
+
+  // IMPORTANT: All strings returned in info must be constant
+  // or static as they will be used _after_ the function returns
+
+  // See https://mflt.io/version-nomenclature for more context
+  *info = (sMemfaultDeviceInfo) {
+    // An ID that uniquely identifies the device in your fleet
+    // (i.e serial number, mac addr, chip id, etc)
+    // Regular expression defining valid device serials: ^[-a-zA-Z0-9_]+$
+    .device_serial = NULL,
+     // A name to represent the firmware running on the MCU.
+    // (i.e "ble-fw", "main-fw", or a codename for your project)
+    .software_type = MYNEWT_VAL(MEMFAULT_DEVICE_INFO_SOFTWARE_TYPE),
+    // The version of the "software_type" currently running.
+    // "software_type" + "software_version" must uniquely represent
+    // a single binary
+    .software_version = MYNEWT_VAL(MEMFAULT_DEVICE_INFO_SOFTWARE_VERS),
+    // The revision of hardware for the device. This value must remain
+    // the same for a unique device.
+    // (i.e evt, dvt, pvt, or rev1, rev2, etc)
+    // Regular expression defining valid hardware versions: ^[-a-zA-Z0-9_\.\+]+$
+    .hardware_version = MYNEWT_VAL(MEMFAULT_DEVICE_INFO_HARDWARE_VERS),
+  };
+}
+//! Last function called after a coredump is saved. Should perform
+//! any final cleanup and then reset the device
+void memfault_platform_reboot(void) {
+  os_reboot(HAL_RESET_REQUESTED);
+  while (1) { } // unreachable
+}
+
+bool memfault_platform_time_get_current(sMemfaultCurrentTime *time) {
+  // !FIXME: If the device tracks real time, update 'unix_timestamp_secs' with seconds since epoch
+  // This will cause events logged by the SDK to be timestamped on the device rather than when they
+  // arrive on the server
+  *time = (sMemfaultCurrentTime) {
+    .type = kMemfaultCurrentTimeType_UnixEpochTimeSec,
+    .info = {
+      .unix_timestamp_secs = 0
+    },
+  };
+
+  // !FIXME: If device does not track time, return false, else return true if time is valid
+  return false;
+}
+
+size_t memfault_platform_sanitize_address_range(void *start_addr, size_t desired_size) {
+  static const struct {
+    uint32_t start_addr;
+    size_t length;
+  } s_mcu_mem_regions[] = {
+    // !FIXME: Update with list of valid memory banks to collect in a coredump
+    {.start_addr = MYNEWT_VAL(MEMFAULT_SANITIZE_START), .length = MYNEWT_VAL(MEMFAULT_SANITIZE_SIZE)},
+  };
+
+  for (size_t i = 0; i < MEMFAULT_ARRAY_SIZE(s_mcu_mem_regions); i++) {
+    const uint32_t lower_addr = s_mcu_mem_regions[i].start_addr;
+    const uint32_t upper_addr = lower_addr + s_mcu_mem_regions[i].length;
+    if ((uint32_t)start_addr >= lower_addr && ((uint32_t)start_addr < upper_addr)) {
+      return MEMFAULT_MIN(desired_size, upper_addr - (uint32_t)start_addr);
+    }
+  }
+
+  return 0;
+}
+
+//! !FIXME: This function _must_ be called by your main() routine prior
+//! to starting an RTOS or baremetal loop.
+int memfault_platform_boot(void) {
```

Comment:
General comment, I think the app should be able to decide which features of the SDK its using based on syscfg (feature flag), and only compile those in here and elsewhere.